RestoreVMFromRestorePoint.json

```diff
@@ -0,0 +1,233 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "<location>"
+        },
+        "networkInterfaceName": {
+            "type": "string",
+            "defaultValue": "<nicName>"
+        },
+        "networkSecurityGroupName": {
+            "type": "string",
+            "defaultValue": "<nsgName>"
+        },
+        "subnetName": {
+            "type": "string",
+            "defaultValue": "<subnetName>"
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "<vnetName>"
+        },
+        "publicIpAddressName": {
+            "type": "string",
+            "defaultValue": "<publicIPName>"
+        },
+        "virtualMachineName": {
+            "type": "string",
+            "defaultValue": "<vmName>"
+        },
+        "virtualMachineComputerName": {
+            "type": "string",
+            "defaultValue": "<computeName>"
+        },
+        "virtualMachineSize": {
+            "type": "string",
+            "defaultValue": "<vmSize>"
+        },
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "<username>"
+        },
+        "adminPassword": {
+            "type": "secureString",
+            "defaultValue": "<password>"
+        },
+        "osdiskRestorePointResourceId": {
+            "type": "string",
+            "defaultValue": "/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroupName>/providers/Microsoft.Compute/restorePointCollections/<RestorePointCollectionName>/restorePoints/<RestorePointName>/diskRestorePoints/<OSDiskRestorePointName>"
+        },
+        "datadiskRestorePointResourceId": {
+            "type": "string",
+            "defaultValue": "/subscriptions/<SubscriptionID>/resourceGroups/<ResourceGroupName>/providers/Microsoft.Compute/restorePointCollections/<RestorePointCollectionName>/restorePoints/<RestorePointName>/diskRestorePoints/<DataDiskRestorePointName>"
+        },
+        "restoredOSDiskName": {
+            "type": "string",
+            "defaultValue": "<restored_osdisk_name>"
+        },
+        "restoredDataDiskName": {
```

Comment:
Customer can change the type (standard, premium, LRS, ZRS) and size (1TB, 4TB etc. ) of the disk right when restoring from diskRestorePoint right? Can we add this to the template?

Reply:
Yes, they can change these params. Do you want me to parameterize these?

Reply:
Yes please.